### PR TITLE
SNOW-1478406: Avoid unnecessary queries on squeeze

### PR DESCRIPTION
**Pull request by sfc-gh-dpetersohn**
1. Which Jira issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   

   Fixes SNOW-1478406

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
      - [ ] If this test skips Local Testing mode, I'm requesting review from @snowflakedb/local-testing
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency
   - [ ] If this is a new feature/behavior, I'm adding the Local Testing parity changes.

3. Please describe how your code solves the related issue.

    Avoid unnecessary length query when `axis=1`

### CODE REVIEW

File: src/snowflake/snowpark/modin/pandas/dataframe.py [hunk 1]
```diff
@@ -2347,15 +2347,15 @@ def squeeze(self, axis: Axis | None = None):
...
-        # dataframe. We should only compute len_index if axis is 0 or None.
-        len_index = len(self)
-        if axis is None and (len_columns == 1 or len_index == 1):
-            return Series(query_compiler=self._query_compiler).squeeze()
-        if axis == 0 and len_index == 1:
-            return Series(query_compiler=self.T._query_compiler)
-        else:
-            return self.copy()
+        if axis in [0, None]:
+            # get_axis_len(0) results in a sql query to count number of rows in current
```
Comment (sfc-gh-yzou) [inline src/snowflake/snowpark/modin/pandas/dataframe.py:2351]: i expect this should impact existing tests and see a query count change, if not, can you add a test for the case to make sure the query count is expected?

Comment (sfc-gh-dpetersohn) [reply src/snowflake/snowpark/modin/pandas/dataframe.py:2351]: Yes I will add a test, I wanted to see what it impacted first 😄

Comment (sfc-gh-dpetersohn) [reply src/snowflake/snowpark/modin/pandas/dataframe.py:2351]: Several tests failed, I have fixed them.

**Review by sfc-gh-nkumar — APPROVED** [review]
Thanks for the fix. Out of curiosity is the new code same as OSS modin or we will have upstream this change?

**Review by sfc-gh-yzou — APPROVED** [review]

**Comment by sfc-gh-dpetersohn** [comment]
@sfc-gh-nkumar this will need to be contributed back to Modin. Link to Modin ticket: https://github.com/modin-project/modin/issues/7315

**Pull request merged**